Essentials/src/com/earth2me/essentials/signs/EssentialsSign.java

```diff
@@ -436,6 +436,30 @@ protected final Trade getTrade(final ISign sign, final int index, final int decr
         }
     }
 
+    /**
+     * Updates the sign cost to the value of the item specified in worth.yml
+     */
+    protected void updateFromWorth(final ISign sign, final IEssentials ess) throws SignException {
+      updateFromWorth(sign, ess, new BigDecimal("1"));
+    }
+    
+    /**
+     * Updates the sign cost to the value of the item specified in worth.yml multiplied by a multiplier factor
+     */
+    protected void updateFromWorth(final ISign sign, final IEssentials ess, final BigDecimal multiplier) throws SignException {
+      final ItemStack stack = getItemStack(getSignText(sign, 2), getIntegerPositive(sign.getLine(1)), ess);
+      final int amount = stack.getAmount();
+      BigDecimal price = ess.getWorth().getPrice(ess, stack);
+      if (price == null || amount == 0) return;
+      price = price.multiply(multiplier).multiply(new BigDecimal(amount));
+      final BigDecimal oldPrice = getMoney(getSignText(sign, 3), ess);
+      if ( oldPrice == null || price.compareTo(oldPrice) != 0) {
+        sign.setLine(3, NumberUtil.shortCurrency(price, ess));
+        sign.updateSign();
+        throw new SignException(tl("priceChanged", oldPrice, price));
```

Comment:
The message will never be sent if you always ignore it.

Reply:
if I test this I see this message. The exception is caught somewhere and shown to the player. I assumed this was how it is supposed to work. Maybe it is a more elegant solution if I catch this somewhere else and display the message there?

Reply:
This is caught on line 64 and 118, I think that's fine